### PR TITLE
python310Packages.bqplot: init at 0.12.36

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Description of changes

Add a package and its dependency.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/bqplot/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  propagatedBuildInputs = [
+    bqscales
+    ipywidgets
+    numpy
+    pandas
+    traitlets
+    traittypes
+  ];
+
+  # no tests in PyPI dist
```
Comment (dotlambda) [inline pkgs/development/python-modules/bqplot/default.nix:39]: Can we use `fetchFromGitHub`?

File: pkgs/development/python-modules/bqscales/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  propagatedBuildInputs = [
+    ipywidgets
+    numpy
+    traitlets
+    traittypes
+  ];
+
+  doCheck = false;
+  # no tests in PyPI dist
```
Comment (dotlambda) [inline pkgs/development/python-modules/bqscales/default.nix:39]: Can we use `fetchFromGitHub`?
```suggestion
  # no tests in PyPI dist
  doCheck = false;
```

File: pkgs/development/python-modules/bqscales/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  disabled = pythonOlder "3.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-C+/GLpqYpePngbn5W0MwvpdmVgFZF7aGHyKMgO5XM90=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+    jupyter-packaging
```
Comment (dotlambda) [inline pkgs/development/python-modules/bqscales/default.nix:28]:
```suggestion
    jupyterlab
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/bqscales/default.nix (hunk 3 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/bqscales/default.nix]: Changed.

File: pkgs/development/python-modules/bqscales/default.nix (hunk 2 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/bqscales/default.nix]: Switched.

File: pkgs/development/python-modules/bqplot/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/python-modules/bqplot/default.nix:39]: Of course I would prefer this, but as usual with these ipy*/jupyter packages it's not immediately obvious how to get the javascript files to be built/included in the right way.

Comment (bcdarwin) [reply pkgs/development/python-modules/bqplot/default.nix]: I added a comment about this.

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 217583` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bqplot</li>
    <li>python310Packages.bqscales</li>
    <li>python311Packages.bqplot</li>
    <li>python311Packages.bqscales</li>
  </ul>
</details>

**Comment by bcdarwin** [comment]
Reverted `jupyterlab` -> `jupyter-packaging`.

**Comment by dotlambda** [comment]
> Reverted `jupyterlab` -> `jupyter-packaging`.

Can you open an upstream issue about their `pyproject.toml` being wrong?

**Pull request merged**